src/lib/routing/lib/slug.ts

```diff
@@ -0,0 +1,38 @@
+import type { SiteLocale } from '@lib/datocms/types';
+
+export const missingSlug = '-';
+export type MaybeSlug = string | undefined;
+
+type LocalizedSlugs = {
+  _allSlugLocales?:
+    | {
+        locale?: SiteLocale | null;
+        value?: string;
+      }[]
+    | null;
+};
```

Comment:
This is a bit of a weird type, but it is what's common between all route fragments with `_allSlugLocales`